addons/product_expiry/models/production_lot.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import datetime
+from openerp import api, fields, models
+
+
+class StockProductionLot(models.Model):
+    _inherit = 'stock.production.lot'
+
+    def _get_date(self, dtype):
+        """Return computed limit date for a given date type."""
+        product_id = self.env.context.get('product_id')
+        if not product_id:
+            date = False
+        else:
+            product = self.env['product.product'].browse(product_id)
+            duration = getattr(product, dtype)
+            # set date to False when no expiry time specified on the product
+            date = duration and (datetime.datetime.today() + datetime.timedelta(days=duration))
+        return date and fields.Datetime.to_string(date) or False
+
+    life_date = fields.Datetime(string='End of Life Date', default=lambda self: self._get_date('life_time'),
+        help='This is the date on which the goods with this Serial Number may become dangerous and must not be consumed.')
+    use_date = fields.Datetime(string='Best before Date', default=lambda self: self._get_date('use_time'),
+        help='This is the date on which the goods with this Serial Number start deteriorating, without being dangerous yet.')
+    removal_date = fields.Datetime(default=lambda self: self._get_date('removal_time'),
+        help='This is the date on which the goods with this Serial Number should be removed from the stock.')
+    alert_date = fields.Datetime(default=lambda self: self._get_date('alert_time'),
+        help="This is the date on which an alert should be notified about the goods with this Serial Number.")
+
+    # Assign dates according to products data
+    @api.model
+    def create(self, vals):
```

Comment:
can not implement onchange method of product_id field instead of over ride create method ?